airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/connector_runner.py

```diff
@@ -90,26 +92,59 @@ def call_read_with_state(self, config, catalog, state, **kwargs) -> List[Airbyte
         self._runs += 1
         volumes = self._prepare_volumes(config, state, catalog)
-        logging.info("Docker run: \n%s\ninput: %s\noutput: %s", cmd, self.input_folder, self.output_folder)
-        try:
-            logs = self._client.containers.run(
-                image=self._image, command=cmd, working_dir="/data", volumes=volumes, network="host", stdout=True, stderr=True, **kwargs
-            )
-        except ContainerError as err:
-            # beautify error from container
-            patched_error = ContainerError(
-                container=err.container, exit_status=err.exit_status, command=err.command, image=err.image, stderr=err.stderr.decode()
-            )
-            raise patched_error from None  # get rid of any previous exception stack
-
-        with open(str(self.output_folder / "raw"), "wb+") as f:
-            f.write(logs)
-
-        for line in logs.decode("utf-8").splitlines():
-            try:
-                yield AirbyteMessage.parse_raw(line)
-            except ValidationError as exc:
-                logging.warning("Unable to parse connector's output %s", exc)
+        logging.debug(f"Docker run {self._image}: \n{cmd}\n"
+                      f"input: {self.input_folder}\noutput: {self.output_folder}")
+
+        container = self._client.containers.run(
+            image=self._image, command=cmd,
+            working_dir="/data", volumes=volumes,
+            auto_remove=True, detach=True,
+            **kwargs,
+        )
+
+        with open(self.output_folder / "raw", "wb+") as f:
+            for line in self.read(container, command=cmd):
+                f.write(line.encode())
+                try:
+                    yield AirbyteMessage.parse_raw(line)
+                except ValidationError as exc:
+                    logging.warning("Unable to parse connector's output %s, error: %s", line, exc)
+
+    @classmethod
+    def read(cls, container: Container, command: str = None, with_ext: bool = True) -> Iterable[str]:
+        """Reads connector's logs per line"""
+        buffer = b''
+        has_exception = False
+        for chunk in container.logs(
+                stdout=True, stderr=True,
+                stream=True, follow=True
+        ):
+            buffer += chunk
+            found = buffer.find(b"\n")
+            if found <= -1:
+                continue
+            line = buffer[:found].decode("utf-8")
+            if has_exception or "Traceback (most recent call last)" in line:
+                has_exception = True
+            else:
+                yield line
+                buffer = buffer[found + 1:]
+        if not has_exception and buffer:
+            yield buffer.decode("utf-8")
+
+        exit_status = container.wait()
+        if exit_status["StatusCode"]:
+            error = buffer.decode("utf-8") or exit_status["Error"]
+            logging.error(f"Docker container was failed, "
```

Comment:
it is not an error

Reply:
A buffer includes the latest traceback because  exit_status["Error"] can be some docker error only.
I made this code part more obvious. Please check again